projects/aws/bottlerocket-bootstrap/pkg/kubeadm/utils.go

```diff
@@ -104,6 +104,31 @@ func getBootstrapFromJoinConfig(path string) (string, string, error) {
 	return "https://" + serverEndpoint, token, nil
 }
 
+func getLocalApiServerReadinessEndpoint() (string, error) {
+	data, err := ioutil.ReadFile("/.bottlerocket/rootfs/etc/kubernetes/manifests/kube-apiserver")
+	if err != nil {
+		return "", errors.Wrap(err, "Error reading ApiServer manifest file")
+	}
+
+	apiServerTree, err := unmarshalIntoMap([]byte(data))
```

Comment:
While this way is more resilient across different versions, we anyways use the v1 import for parsing Pod spec already. That would make this a bit easier to get to the scheme/host/port. What are your thoughts on using the v1.Pod struct here. An example [here](https://github.com/aws/eks-anywhere-build-tooling/blob/fcb1eaf1e2425cae4c1d73bd590e6a7f29b535b2/projects/aws/bottlerocket-bootstrap/pkg/utils/waiters.go#L178-L196)

Reply:
There is a helper [here](https://github.com/aws/eks-anywhere-build-tooling/blob/fcb1eaf1e2425cae4c1d73bd590e6a7f29b535b2/projects/aws/bottlerocket-bootstrap/pkg/utils/parsers.go#L50) you can use

Reply:
Good idea, I also prefer using a proper type for this. 
Thanks for pointing that out!